eng/scripts/Invoke-DependencyCheck.ps1

```diff
@@ -0,0 +1,77 @@
+Param(
+    [string] $newPackageDirectory
+)
+
+. (Join-Path $PSScriptRoot .. common scripts common.ps1)
+
+$ignoreCheck = "sdk/synapse/azartifacts", "sdk/template/aztemplate", "sdk/azcore", "sdk/azidentity"
```

Comment:
Why do we want to ignore these? Is this temporary and expected to be fixed later? If so can we file an issue and add a comment here linking to that issue.

Reply:
Regardless I'd rather not hardcode skips into our scripts. It's easy to completely forget about them. I think it would also be better to have these as sdk-level opt-outs, rather than in a centralized place, e.g. passing them down through the `ci.yml` files as parameters in the respective service directories.

Reply:
I've removed most of ignore package here except one: sdk/synapse/azartifacts. This package is a deprecated package that  code has not deleted yet. So I need to ignore this package in the dependency check.

Reply:
We might consider some other way of filtering this out because I expect this ignore list will be left here forever now even after that project is removed. At any rate this isn't a blocker but it also isn't the ideal way of excluding things.

Reply:
I've changed the implementation for detecting deprecated packages. The deprecated packages in go will have `retract` section in `go.mod` file. So I'll skip the package by whether the package version is in the `retract` section.